data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/accumulator/LocalFileBufferTest.java

```diff
@@ -0,0 +1,79 @@
+package org.opensearch.dataprepper.plugins.sink.accumulator;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.s3.S3Client;
+import java.io.IOException;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@ExtendWith(MockitoExtension.class)
+class LocalFileBufferTest {
+
+    public static final String BUCKET_NAME = UUID.randomUUID().toString();
+    @Mock
+    private S3Client s3Client;
+    private LocalFileBuffer localFileBuffer;
+
+    @Test
+    void test_with_write_event_into_buffer() throws IOException {
+        localFileBuffer = new LocalFileBuffer();
+
+        while (localFileBuffer.getEventCount() < 55) {
+            localFileBuffer.writeEvent(generateByteArray());
+        }
+        assertThat(localFileBuffer.getSize(), greaterThan(1l));
+        assertThat(localFileBuffer.getEventCount(), equalTo(55));
+        assertThat(localFileBuffer.getDuration(), greaterThanOrEqualTo(0L));
```

Comment:
You should be able to assert the exact duration here.

```
assertThat(localFileBuffer.getDuration(), equalTo(55 * 1000L));
```

Reply:
Addressed.